src/plugins/data/common/search/session/types.ts

```diff
@@ -38,10 +49,53 @@ export interface ISessionService {
   /**
    * Restores existing session
    */
-  restore: (sessionId: string) => void;
+  restore: (sessionId: string) => Promise<any>;
 
   /**
    * Clears the active session.
    */
   clear: () => void;
+
+  /**
+   * Saves a session
+   */
+  save: (name: string, url: string) => Promise<any>;
+
+  /**
+   * Gets a saved session
+   */
+  get: (sessionId: string) => Promise<any>;
+
+  /**
+   * Gets a list of saved sessions
+   */
+  find: (options: SearchSessionFindOptions) => Promise<any>;
+
+  /**
+   * Updates a session
+   */
+  update: (
+    sessionId: string,
+    attributes: Partial<BackgroundSessionSavedObjectAttributes>
+  ) => Promise<any>;
```

Comment:
Would would be the proper return types for this CRUD methods?
I can add in follow up client side PR, just not sure what are those.